text/3723-static-function-argument-unpacking.md

```diff
@@ -0,0 +1,753 @@
+- Feature Name: `static_function_argument_unpacking`
+- Start Date: 2024-10-30
+- RFC PR: [rust-lang/rfcs#3723](https://github.com/rust-lang/rfcs/pull/3723)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC adds call-site unpacking of tuples, tuple structs, and fixed-size arrays, using `...expr` within the function call's parentheses as a shorthand for passing arguments. The full contents of these collections with known sizes are unpacked directly as the next arguments of a function call, desugaring into the corresponding element accesses during compilation.
+
+# Motivation
+[motivation]: #motivation
+
+Argument unpacking reduces the verbosity and increases the ergonomics of Rust, it:
+
+- Improves code writing ergonomics by removing the need for repetitive, unneeded intermediate steps.
+- Allows more concise code in terms of number of lines and, occasionally, line length.
+- Allows reducing the number of named local variables in scope.
```

Comment:
Since you already have a tuple/array of this, you could simply pass `tuple.0, tuple.1, tuple.2` to the argument list instead of `...tuple`. I don't see how this avoids additional local variables, unless you are referring to avoiding naming `tuple` itself, which would only be the case when the output of another function happens to coincide with (a subsequence of) the inputs of the called function, in which case, again, should have been made its own struct.